modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryHelper.java

```diff
@@ -64,50 +63,46 @@
 
 			evaluableDDMFormFieldNames.addAll(
 				getReferencedFieldNamesByExpression(
-					ddmFormField.getVisibilityExpression(), ddmFormFieldNames));
+					ddmFormField.getVisibilityExpression()));
 		}
 
-		return evaluableDDMFormFieldNames;
+		ddmFormFieldNames.retainAll(evaluableDDMFormFieldNames);
+
+		return ddmFormFieldNames;
```

Comment:
@cleydyr, I visited the [retainAll Java API](https://docs.oracle.com/javase/8/docs/api/java/util/Set.html#retainAll-java.util.Collection-) and I realized that this method returns a boolean, however I didn't get the purpose of its use here, can you please clarify?

Reply:
Renatonho, the return value can be disregarded as it only indicates whether the collection where the `retainAll` method was called was changed as a result of this method call. What we're interested is in changing the ddmFormFieldNames collection and returning the changed collection.

The retainAll method can be used to perform an intersection operation between sets. "only the elements in this set that are contained in the specified collection" is the mathematical definition of intersection between this set and the specified collection (which may be a set or not).

evaluableDDMFormFieldNames (in this commit, it is renamed in a later commit) is a set of arguments from all expressions. So if we get an intersection between the set of form field names and all the arguments from the expression, we're getting the _evaluable_ form field names as before, only using a different approach.